runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -601,7 +601,7 @@ private Object applyUpdateUnsafe(SMREntry entry, long timestamp) {
                 // This entry actually resets the object. So here
                 // we can safely get a new instance, and add the
                 // previous instance to the undo log.
-                entry.setUndoRecord(object.getContext(context));
+                entry.setUndoRecord(object);
```

Comment:
is it possible to have a test for this logic which demonstrates that this logic is ok and correct?

Reply:
This bug was caught as part of this PR: https://github.com/CorfuDB/CorfuDB/pull/2254

So, there is a test, but it is currently leveraging SMRMap and not CorfuTable.